joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1007,6 +1023,14 @@ controller_interface::CallbackReturn JointTrajectoryController::on_activate(
   }
   rt_is_holding_.writeFromNonRT(true);
 
+  // update gains of PID controller
+  if (use_closed_loop_pid_adapter_)
+  {
+    trajectory_msgs::msg::JointTrajectory traj;
+    traj.points.push_back(state_current_);
```

Comment:
@christophfroehlich  Maybe also adding the `state_desired_` would help

Reply:
`computeGains` takes a full trajectory, here it is only initialized with the same points as used for `set_hold_position`

As mentioned in the description, an update for dynamic reconfigure is missing. 

But there is also a call missing for receiving a new trajectory, I'll push an update here.

Reply:
I pushed a new update